mlrun/api/api/endpoints/artifacts_v2.py

```diff
@@ -0,0 +1,280 @@
+    logger.debug("Creating artifact", project=project, key=key, tag=tag, iter=iteration)
+    await mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        mlrun.common.schemas.AuthorizationResourceTypes.artifact,
+        project,
+        key,
+        mlrun.common.schemas.AuthorizationAction.store,
+        auth_info,
+    )
+    artifact_uid = await run_in_threadpool(
+        mlrun.api.crud.Artifacts().create_artifact,
+        db_session,
+        key,
+        artifact.dict(exclude_none=True),
+        tag,
+        iteration,
+        tree,
+        project,
+    )
+    return await run_in_threadpool(
+        mlrun.api.crud.Artifacts().get_artifact,
+        db_session,
+        key,
+        tag,
+        iteration,
+        project,
+        tree=tree,
+        object_uid=artifact_uid,
+    )
+
+
+@router.put("/projects/{project}/artifacts/{key:path}")
+async def store_artifact(
+    project: str,
+    artifact: mlrun.common.schemas.Artifact,
+    key: str,
+    tree: str = None,
+    tag: str = None,
+    iter: int = 0,
+    object_uid: str = None,
+    auth_info: mlrun.common.schemas.AuthInfo = Depends(deps.authenticate_request),
+    db_session: Session = Depends(deps.get_db_session),
+):
+    await run_in_threadpool(
+        mlrun.api.utils.singletons.project_member.get_project_member().ensure_project,
+        db_session,
+        project,
+        auth_info=auth_info,
+    )
+
+    logger.debug("Storing artifact", project=project, key=key, tag=tag, iter=iter)
+    await mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        mlrun.common.schemas.AuthorizationResourceTypes.artifact,
+        project,
+        key,
+        mlrun.common.schemas.AuthorizationAction.store,
+        auth_info,
+    )
+    artifact_uid = await run_in_threadpool(
+        mlrun.api.crud.Artifacts().store_artifact,
+        db_session,
+        key,
+        artifact.dict(exclude_none=True),
+        object_uid,
+        tag,
+        iter,
+        project,
+        tree,
+    )
+    return await run_in_threadpool(
+        mlrun.api.crud.Artifacts().get_artifact,
+        db_session,
+        key,
+        tag,
+        iter,
+        project,
+        tree=tree,
+        uid=artifact_uid,
+    )
+
+
+@router.get("/projects/{project}/artifacts")
+async def list_artifacts(
+    project: str = None,
+    name: str = None,
+    tag: str = None,
+    kind: str = None,
+    category: mlrun.common.schemas.ArtifactCategories = None,
+    labels: List[str] = Query([], alias="label"),
+    iter: int = Query(None, ge=0),
+    tree: str = None,
+    best_iteration: bool = Query(False, alias="best-iteration"),
+    format_: ArtifactsFormat = Query(ArtifactsFormat.full, alias="format"),
+    auth_info: mlrun.common.schemas.AuthInfo = Depends(deps.authenticate_request),
+    db_session: Session = Depends(deps.get_db_session),
+):
+    if project is None:
+        project = config.default_project
```

Comment:
it seems to me that project can never be none

Reply:
Right, I took it from the original artifacts but it was probably because `/artifacts` is still supported.